---
title: "Vignette_mobiCD"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Vignette_mobiCD}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

```{r setup}
library(mobiCD)
```

# Description

# Individual function explanation

## generate_dna()

The first function called generate_dna

## match_replace()

## split_into_codons()

## tRNA_rep(codons)

tRNA_rep() function is used to correspond a codon to an amino-acid, just like a tRNA would do in the Central Dogma of molecular biology. We have included a table of codons with their corresponding amino-acids in the package. So, when you call the function with a 3 RNA nucleotide sequence, it tries to match it with an amino-acid from that table. If a big sequence of codons are given as an input, a protein sequence will be returned.

## plot_content(protein)

plot_content() function gets as input a protein and returns a plot showing the population of each amino-acid in this protein sequence. As a first step, it splits the sequence to separate amino-acids. Then, it creates a new table containing the counts for every amino-acid found previously. Finally, it generates a bar plot with all

# Example of package usage

```{r}
dna_seq <- generate_dna(60) 
rna_seq <- match_replace(dna_seq)
codons <- split_into_codons(rna_seq)
protein <- tRNA_rep(codons)
plot_content(protein)

  
```
